graylog2-server/src/main/java/org/graylog2/indexer/indices/Indices.java

```diff
@@ -394,4 +398,17 @@ public void optimizeIndex(String index) {
 
         c.admin().indices().optimize(or).actionGet();
     }
+
+    public void waitForRecovery(String index) {
+        RecoveryResponse response;
+
+        do {
+            LOG.warn("Waiting for index recovery for index {}", index);
+
+            final ActionFuture<RecoveryResponse> request = c.admin().indices().recoveries(new RecoveryRequest(index));
```

Comment:
I think a `ClusterHealthRequest` for `index` with `waitForStatus` YELLOW or GREEN might be more useful here (https://www.elastic.co/guide/en/elasticsearch/reference/1.5/cluster-health.html#request-params) and doesn't require us to poll ourselves.